Add points-on-line eval

aaronsmithtv opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. We encourage partial PR's with ~5-10 example that we can then run the evals on and share the results with you so you know how your eval does with GPT-4 before writing all 100 examples.

## Eval details 📑
### Eval name
Points On Line

### Eval description

100 sets of vector coordinates in the form of `(x, y, z), (x, y, z)`, with an ideal centre coordinate. The coordinates have a random start position of `(-1, -1, -1)` to `(1, 1, 1)` and a furthest maximum distance from origin per-component of 2. All positions are in steps of 0.1 for ease of readability and human understanding.

### What makes this a useful eval?

This eval helps gain insight on a GPT model's ability to understand a coordinate space. This is historically a subject that LLMs have been poor in, and provides a reliable, predictable benchmark for whether a model can understand the context of positions within a coordinate space.

gpt-3.5-turbo has an average level of performance, with 0.3 total accuracy on this eval:

```console
[2023-03-17 12:38:05,659] [record.py:320] Final report: {'accuracy': 0.3}. Logged to /tmp/evallogs/230317123702XGUGPDEY_gpt-3.5-turbo_points-on-line.jsonl
[2023-03-17 12:38:05,659] [oaieval.py:209] Final report:
[2023-03-17 12:38:05,659] [oaieval.py:211] accuracy: 0.3

```

gpt-4 has a much greater level of accuracy at 0.86, indicating its improved ability to understand vector coordinates:

```console
[2023-03-17 12:51:53,570] [record.py:320] Final report: {'accuracy': 0.86}. Logged to /tmp/evallogs/230317125110ELCUGZXE_gpt-4_points-on-line.jsonl
[2023-03-17 12:51:53,571] [oaieval.py:209] Final report:
[2023-03-17 12:51:53,571] [oaieval.py:211] accuracy: 0.86
```

However, the results for both are very consistent, showing that there may be unusual cases where certain vector positions are impossible for either GPT model to return correctly.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] Include at least 100 high quality examples (it is okay to only contribute 5-10 meaningful examples and have us test them with GPT-4 before adding all 100)

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> These evals come with a generator script that can create new coordinate datasets very quickly. It can also be expanded to account for future, more difficult scopes of this test, such as larger distances, greater floating point deviation, and total numbers of points to calculate in a space.

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.jsonl`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields in the evals PR form
- [x] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(1.0, 0.5, -0.8), (1.0, 1.5, -0.2)"}], "ideal": "(1.0, 0.5, -0.8), (1.0, 1.0, -0.5), (1.0, 1.5, -0.2)"}
{"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.3, 0.0, 0.5), (0.9, 0.2, 2.1)"}], "ideal": "(0.3, 0.0, 0.5), (0.6, 0.1, 1.3), (0.9, 0.2, 2.1)"}
{"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.5, 1.0, 1.0), (-1.5, -1.0, 0.2)"}], "ideal": "(0.5, 1.0, 1.0), (-0.5, 0.0, 0.6), (-1.5, -1.0, 0.2)"}
{"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(-0.6, 0.0, 0.6), (0.0, -1.8, 2.4)"}], "ideal": "(-0.6, 0.0, 0.6), (-0.3, -0.9, 1.5), (0.0, -1.8, 2.4)"}
{"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.4, -0.9, 0.4), (1.4, -0.3, -0.8)"}], "ideal": "(0.4, -0.9, 0.4), (0.9, -0.6, -0.2), (1.4, -0.3, -0.8)"}
  ```
</details>

SierotkaM commented:
Sam rozgryzam o co w tym chodzi że dwa dni mi to zajmnie  także równinie
dobrze możecie na Atlantydę pisać skóry siła wyższa . Co 25 będę dopisywał
coś mam nadzieję że nic niedobrego się nie dzieje jak tego nie rozumiem ,😟

pt., 17 mar 2023, 14:14 użytkownik Aaron Smith ***@***.***>
napisał:

> Thank you for contributing an eval! ♥️
>
> 🚨 Please make sure your PR follows these guidelines, *failure to follow
> the guidelines below will result in the PR being closed automatically*.
> Note that even if the criteria are met, that does not guarantee the PR will
> be merged nor GPT-4 access granted. 🚨
>
> *PLEASE READ THIS*:
>
> In order for a PR to be merged, it must fail on GPT-4. We are aware that
> right now, users do not have access, so you will not be able to tell if the
> eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in
> mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will
> likely reject since GPT-4 is already capable of completing the task.
>
> We plan to roll out a way for users submitting evals to see the eval
> performance on GPT-4 soon. Stay tuned! Until then, you will not be able to
> see the eval performance on GPT-4. We encourage partial PR's with ~5-10
> example that we can then run the evals on and share the results with you so
> you know how your eval does with GPT-4 before writing all 100 examples.
> Eval details 📑 Eval name
>
> Points On Line
> Eval description
>
> 100 sets of vector coordinates in the form of (x, y, z), (x, y, z), with
> an ideal centre coordinate. The coordinates have a random start position of (-1,
> -1, -1) to (1, 1, 1) and a furthest maximum distance from origin
> per-component of 2. All positions are in steps of 0.1 for ease of
> readability and human understanding.
> What makes this a useful eval?
>
> This eval helps gain insight on a GPT model's ability to understand a
> coordinate space. This is historically a subject that LLMs have been poor
> in, and provides a reliable, predictable benchmark for whether a model can
> understand the context of positions within a coordinate space.
>
> gpt-3.5-turbo has an average level of performance, with 0.3 total accuracy
> on this eval:
>
> [2023-03-17 12:38:05,659] [record.py:320] Final report: {'accuracy': 0.3}. Logged to /tmp/evallogs/230317123702XGUGPDEY_gpt-3.5-turbo_points-on-line.jsonl[2023-03-17 12:38:05,659] [oaieval.py:209] Final report:[2023-03-17 12:38:05,659] [oaieval.py:211] accuracy: 0.3
>
> gpt-4 has a much greater level of accuracy at 0.86, indicating its
> improved ability to understand vector coordinates:
>
> [2023-03-17 12:51:53,570] [record.py:320] Final report: {'accuracy': 0.86}. Logged to /tmp/evallogs/230317125110ELCUGZXE_gpt-4_points-on-line.jsonl[2023-03-17 12:51:53,571] [oaieval.py:209] Final report:[2023-03-17 12:51:53,571] [oaieval.py:211] accuracy: 0.86
>
> However, the results for both are very consistent, showing that there may
> be unusual edge cases where certain vector positions are impossible for
> either GPT model to return correctly.
> Criteria for a good eval ✅
>
> Below are some of the criteria we look for in a good eval. In general, we
> are seeking cases where the model does not do a good job despite being
> capable of generating a good response (note that there are some things
> large language models cannot do, so those would not make good evals).
>
> Your eval should be:
>
>    - Thematically consistent: The eval should be thematically consistent.
>    We'd like to see a number of prompts all demonstrating some particular
>    failure mode. For example, we can create an eval on cases where the model
>    fails to reason about the physical world.
>    - Contains failures where a human can do the task, but either GPT-4 or
>    GPT-3.5-Turbo could not.
>    - Includes good signal around what is the right behavior. This means
>    either a correct answer for Basic evals or the Fact Model-graded eval,
>    or an exhaustive rubric for evaluating answers for the Criteria
>    Model-graded eval.
>    - Include at least 100 high quality examples (it is okay to only
>    contribute 5-10 meaningful examples and have us test them with GPT-4 before
>    adding all 100)
>
> If there is anything else that makes your eval worth including, please
> document it below.
> Unique eval value
>
> These evals come with a generator script that can create new coordinate
> datasets very quickly. It can also be expanded to account for future, more
> difficult scopes of this test, such as larger distances, greater floating
> point deviation, and total numbers of points to calculate in a space.
>
> Eval structure 🏗️
>
> Your eval should
>
>    - Check that your data is in evals/registry/data/{name}
>    - Check that your yaml is registered at
>    evals/registry/evals/{name}.jsonl
>    - Ensure you have the right to use the data you submit via this eval
>
> (For now, we will only be approving evals that use one of the existing
> eval classes. You may still write custom eval classes for your own cases,
> and we may consider merging them in the future.)
> Final checklist 👀 Submission agreement
>
> By contributing to Evals, you are agreeing to make your evaluation logic
> and data under the same MIT license as this repository. You must have
> adequate rights to upload any data used in an Eval. OpenAI reserves the
> right to use this data in future service improvements to our product.
> Contributions to OpenAI Evals will be subject to our usual Usage Policies (
> https://platform.openai.com/docs/usage-policies).
>
>    - I agree that my submission will be made available under an MIT
>    license and complies with OpenAI's usage policies.
>
> Email address validation
>
> If your submission is accepted, we will be granting GPT-4 access to a
> limited number of contributors. Access will be given to the email address
> associated with the merged pull request.
>
>    - I acknowledge that GPT-4 access will only be granted, if applicable,
>    to the email address used for my merged pull request.
>
> Limited availability acknowledgement
>
> We know that you might be excited to contribute to OpenAI's mission, help
> improve our models, and gain access to GPT-4. However, due to the
> requirements mentioned above and high volume of submissions, we will not be
> able to accept all submissions and thus not grant everyone who opens a PR
> GPT-4 access. We know this is disappointing, but we hope to set the right
> expectation before you open this PR.
>
>    - I understand that opening a PR, even if it meets the requirements
>    above, does not guarantee the PR will be merged nor GPT-4 access granted.
>
> Submit eval
>
>    - I have filled out all required fields in the evals PR form
>    - (Ignore if not submitting code) I have run pip install pre-commit;
>    pre-commit install and have verified that black, isort, and autoflake
>    are running when I commit and push
>
> Failure to fill out all required fields will result in the PR being closed.
> Eval JSON data
>
> Since we are using Git LFS, we are asking eval submitters to add in as
> many Eval Samples (at least 5) from their contribution here:
> View evals in JSON Eval
>
> {"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(1.0, 0.5, -0.8), (1.0, 1.5, -0.2)"}], "ideal": "(1.0, 0.5, -0.8), (1.0, 1.0, -0.5), (1.0, 1.5, -0.2)"}
> {"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.3, 0.0, 0.5), (0.9, 0.2, 2.1)"}], "ideal": "(0.3, 0.0, 0.5), (0.6, 0.1, 1.3), (0.9, 0.2, 2.1)"}
> {"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.5, 1.0, 1.0), (-1.5, -1.0, 0.2)"}], "ideal": "(0.5, 1.0, 1.0), (-0.5, 0.0, 0.6), (-1.5, -1.0, 0.2)"}
> {"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(-0.6, 0.0, 0.6), (0.0, -1.8, 2.4)"}], "ideal": "(-0.6, 0.0, 0.6), (-0.3, -0.9, 1.5), (0.0, -1.8, 2.4)"}
> {"input": [{"role": "system", "content": "There are three points on a line between the coordinates provided. Return only the coordinates of the points on the line. The response should also be as concise as possible, with no extra information other than the output. The format of the response should be (x, y, z), (x, y, z), (x, y, z)"}, {"role": "user", "content": "(0.4, -0.9, 0.4), (1.4, -0.3, -0.8)"}], "ideal": "(0.4, -0.9, 0.4), (0.9, -0.6, -0.2), (1.4, -0.3, -0.8)"}
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/openai/evals/pull/288
> Commit Summary
>
>    - 0ae5762
>    <https://github.com/openai/evals/pull/288/commits/0ae5762a86bc92e0a33e4634948e0f0494a41a98>
>    Add points-on-line eval
>
> File Changes
>
> (2 files <https://github.com/openai/evals/pull/288/files>)
>
>    - *A* evals/registry/data/points_on_line/points_on_line.jsonl
>    <https://github.com/openai/evals/pull/288/files#diff-3e4a816defa5332b92cdd9bbe91554c860f386f6c8b1d8002a0ee0623f29fbd8>
>    (3)
>    - *A* evals/registry/evals/points-on-line.yaml
>    <https://github.com/openai/evals/pull/288/files#diff-39a047b319e120d326d6d2d21664681bcde266e29ebde5cb387c7eed6a3a4b80>
>    (9)
>
> Patch Links:
>
>    - https://github.com/openai/evals/pull/288.patch
>    - https://github.com/openai/evals/pull/288.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openai/evals/pull/288>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6PJHHWMXH7CUISQNLRSDHDW4RPSPANCNFSM6AAAAAAV6QSSVI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

SierotkaM commented:
Chyba coś nie tak bo na czerwono się świeci  skóry jakby niedoborze

joe-at-openai commented:
Hi, thanks for your submission!

The task as stated is somewhat ambiguous. Using this instruction would make it unambiguous: "You will be provided with the end points of a line in 3 dimensions. Provide the midpoint of this line formatted as (x, y, z)." Please change the answer to require only the midpoint.

usama-openai commented:
Closing the PR due to inactivity; please reopen if you get a chance to address comments.

aaronsmithtv commented:
Hey @joe-at-openai and @usama-openai , many thanks for the helpful suggestions in improving my eval! I've taken a fresh look at the prompt and results, and I've added a new set of samples for you to take a look at.

`gpt-3.5-turbo` fails to provide answers that would satisfy the `Match` class, so I'm now using `Include`. I've also added some extra complexity, since gpt-4 seemed to do incredibly well on the simpler math with 1 decimal!

Here's the two accuracy reports:

```shell
[2023-06-03 01:20:18,964] [record.py:341] Final report: {'accuracy': 0.0}. Logged to /tmp/evallogs/230603001824VWSNJZEG_gpt-3.5-turbo_points-on-line.jsonl
[2023-06-03 01:20:18,964] [oaieval.py:147] Final report:
[2023-06-03 01:20:18,964] [oaieval.py:149] accuracy: 0.0
```

```shell
[2023-06-03 01:21:47,663] [record.py:341] Final report: {'accuracy': 0.66}. Logged to /tmp/evallogs/23060300212233RTRLC7_gpt-4_points-on-line.jsonl
[2023-06-03 01:21:47,663] [oaieval.py:147] Final report:
[2023-06-03 01:21:47,663] [oaieval.py:149] accuracy: 0.66
```

Here you can see that gpt-4 seems to do significantly better than 3.5.

Here's an example of 5 out of the 100 evals from the new eval set:

```json
{"input": [{"role": "system", "content": "You will be provided with the end points of a line in 3 dimensions. Please calculate and return only the midpoint of this line, in this format: (x, y, z)"}, {"role": "user", "content": "(4.10, -4.98, -6.99), (14.28, -23.12, 4.89)"}], "ideal": "(9.19, -14.05, -1.05)"}
{"input": [{"role": "system", "content": "You will be provided with the end points of a line in 3 dimensions. Please calculate and return only the midpoint of this line, in this format: (x, y, z)"}, {"role": "user", "content": "(-1.98, -5.97, -9.94), (-21.94, -19.87, 2.02)"}], "ideal": "(-11.96, -12.92, -3.96)"}
{"input": [{"role": "system", "content": "You will be provided with the end points of a line in 3 dimensions. Please calculate and return only the midpoint of this line, in this format: (x, y, z)"}, {"role": "user", "content": "(2.09, 9.92, 1.06), (4.13, 27.90, -5.14)"}], "ideal": "(3.11, 18.91, -2.04)"}
{"input": [{"role": "system", "content": "You will be provided with the end points of a line in 3 dimensions. Please calculate and return only the midpoint of this line, in this format: (x, y, z)"}, {"role": "user", "content": "(7.07, -1.05, 0.94), (-13.07, -11.17, 17.10)"}], "ideal": "(-3.00, -6.11, 9.02)"}
{"input": [{"role": "system", "content": "You will be provided with the end points of a line in 3 dimensions. Please calculate and return only the midpoint of this line, in this format: (x, y, z)"}, {"role": "user", "content": "(6.90, 4.92, 1.93), (0.74, -11.14, -4.11)"}], "ideal": "(3.82, -3.11, -1.09)"}
```

Thanks again for taking the time to look at my eval and help me get it right :)
